### PR TITLE
SWAP-1246 Use shared validation logic for use profile

### PR DESCRIPTION
**Pull request by peter-asztalos**
## Description

User profile validation updated, validation rules existing only on the FE side but missing on the BE added

### CODE REVIEW

File: packages/validation/src/User/index.ts [hunk 1]
```diff
@@ -65,31 +74,43 @@ export const createUserValidationSchema = Yup.object().shape({
...
-  telephone_alt: Yup.string()
-    .min(2, 'telephone must be at least 2 characters')
-    .max(30, 'telephone must be at most 20 characters')
-    .matches(phoneRegExp, 'Phone number is not valid')
-    .notRequired(),
+    .min(2)
+    .max(30)
+    .matches(phoneRegExp, 'telephone number is not valid')
+    .required(),
+  telephone_alt: Yup.string().max(50),
```
Comment (martin-trajanovski) [inline packages/validation/src/User/index.ts:90]: Isn't it better to use same check for telephone and telephone_alt but just make it as not required or as it was previously inside update schema?

Comment (peter-asztalos) [reply packages/validation/src/User/index.ts:90]: Well, I think we should be more flexible with telephone alt, for example, the default user officer user has such a telephone alt that fails the validation but looks valid number.
Phone number validation, in general, is very hard, as we only let people register who has ORCID account we shouldn't be very strict.
For regular phone numbers removing whitespace, dash, parentheses, and checking only for numbers should be also sufficient I think (e.g. I can't register with my own phone number unless I remove some formating although it's still a valid number.)

Comment (martin-trajanovski) [reply packages/validation/src/User/index.ts:90]: Maybe for phone numbers we should use some special inputs like https://github.com/bl00mber/react-phone-input-2 or https://github.com/alexplumb/material-ui-phone-number where we can support formatting and easier validation.

Comment (peter-asztalos) [reply packages/validation/src/User/index.ts:90]: I personally wouldn't go that far, it's not really important and it's unlikely that a scientist would spam the phone number with something stupid in the user office app.

**Review by martin-trajanovski — APPROVED** [review]

**Pull request merged**